app/services/imo_api_service.rb

```diff
@@ -42,27 +65,55 @@ def execute_post(url, data)
       .basic_auth(user: IMO_USERNAME, pass: IMO_PASSWORD)
       .post(url, data)
   rescue HTTP::Error => e
-    raise Error.new("Error while calling the IMO API", path: url, exception_message: e)
+    raise Error.new("Error while calling the Imo API", path: url, exception_message: e)
   end
 
-  def process_response(response, url)
+  def process_response(response, url, action)
     case response.status
-    when 200 then :invited
+    when 200
+      body = JSON.parse(response.body)
+      body_status = body.dig("response", "status")
+      return :invited if body_status == "success" && action == "invitation"
```

Comment:
The `action` is just what we, the callers, are attempting to do with IMO - right?  I can see how that is useful for the instrumentation below, but we don't really need it for the response handling.  

I would suggest returning `true` here for success, and let the calling method set the correct state based on that.  That way we won't need these two cases for the success case.  Just something like:

```suggestion
      return true if status == "success"
```

Then callers can do either

`ImoAuthorization.create(status: :invited)` or `imo_authorization.update(status: :subscribed)`  .  Or better yet, use the enum methods where possible...for example, `imo_authorization.subscribed!`

Reply:
The action is also being used by Statsd. I could remove the statsd logging or make it non-specific to the action. But I think returning true makes the logic more complex. The reason I didn't use the enum methods is because there are multiple return values from this function that can all be used to update the enum value. So if the user has no account, we update the ImoAuthorization with `:no_imo_account`. If you still think this should be changed, maybe we should zoom for a couple of minutes to avoid any confusion.

Reply:
Actually, after working ahead a little, I've decided I will change this. The handling will have to change after the callbacks are added anyhow. If you don't mind turning a blind eye to this for the moment, I'll change it in the next PR, which is branched off of this.

Reply:
Okay, sounds good, I can follow along there.